File: particle-wallet/android/src/main/java/com/particlewallet/model/ReactErrorMessage.java
package com.particlewallet.model;

import com.connect.common.model.ConnectError;
import com.google.gson.annotations.SerializedName;

public class ReactErrorMessage {
    public static final int ExceptionCode = 10000;
    @SerializedName("code")
    public int code;

    @SerializedName("message")
    public String message;


    public ReactErrorMessage(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public ReactErrorMessage() {
    }

    public static ReactErrorMessage exceptionMsg(String message) {
        return new ReactErrorMessage(ExceptionCode, message);
    }

    public static ReactErrorMessage parseConnectError(ConnectError error) {
        return new ReactErrorMessage(error.getCode(), error.getMessage());
    }
}

File: particle-aa/android/src/main/java/com/particleaa/model/ReactErrorMessage.kt
package com.particleaa.model

import com.google.gson.annotations.SerializedName

class ReactErrorMessage {
    @SerializedName("code")
    var code = 0

    @SerializedName("message")
    var message: String? = null

    constructor(code: Int, message: String?) {
        this.code = code
        this.message = message
    }

    constructor() {}

    companion object {
        const val ExceptionCode = 10000
        fun exceptionMsg(message: String?): ReactErrorMessage {
            return ReactErrorMessage(ExceptionCode, message)
        }
    }
}
